pkg/crypto/crypto_test.go

```diff
@@ -126,6 +128,49 @@ func TestEncryptDecrypt(t *testing.T) {
 	})
 }
 
+func TestUpgradePathEncryptDecrypt(t *testing.T) {
+	simplePassphrase := []byte("secret key")
+
+	testCases := []struct {
+		name           string
+		input          string
+		encryptKey     []byte
+		decryptKey     []byte
+		expectedOut    string
+		expectedEncErr bool
+		expectedDecErr bool
+	}{
+		// error cases
+		{
+			name:           "decrypt previous version encrypted key",
+			input:          "Administrator",
+			encryptKey:     simplePassphrase,
+			decryptKey:     simplePassphrase,
+			expectedOut:    "Administrator",
+			expectedEncErr: false,
+			expectedDecErr: false,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			cipherText, err := getEncryptedString(test.input, test.encryptKey)
```

Comment:
This could be simplified to remove the need for the helper `getEncryptedString`. I don't have a strong opinion either way, just worth thinking about
```suggestion
			cipherText, err := startTag + "\n\n" + encrypt(test.input, test.encryptKey) + "\n" + endTag 
```

Reply:
I am a +1 if we can remove code 😃

Reply:
Initially I had it simple like that. but Instead of creating old format by inserting values, i prefer to use same old way to generate code so that its guaranteed correct format so that our test case is 100% test the old format. 
would like to keep it this way. :)

Reply:
Then please add a doc to `getEncryptedString` explaining this. Otherwise it looks like overly complicated code for no reason. I also suggest naming the function `generateBuggyEncryptedString()` in addition.